applications/dashboard/controllers/api/UsersApiController.php

```diff
@@ -218,7 +224,7 @@ public function getGuestFragment() {
      * Get a list of users, filtered by username.
      *
      * @param array $query
-     * @return array
+     * @return Data
      */
     public function index_byNames(array $query) {
```

Comment:
If we're updating this we might as well make it and explicit function return type. I'm still quite confused though about why we sometimes return Data and other times arrays. It seems inconsistent.
```suggestion
    public function index_byNames(array $query): Data {
```

Reply:
While I'm okay with this particular change, I don't think this should be standard. I made a very minor update to an existing method's documentation, because my IDE was complaining about it. You're now proposing I change the function signature and increase the potential for a fatal error. Again, in this particular case, it shouldn't be a big deal, but this isn't a normal "while we're in there" type of change.

> I'm still quite confused though about why we sometimes return Data and other times arrays. It seems inconsistent.

You only return a `Data` object when you want to modify more of the response than just the body. If you return an array, that's the body of the response. If you return a `Data` object, you can modify the response codes and other "meta'" information about that response. In this particular case, we're adding "Link" headers for pagination.

Reply:
I get what you saying but seeing a short method returning a `new Data()` object it makes sense to me. It's not like this is some ancient legacy code here. The whole thing was written in the last 2 years. We should be able to know what are methods return confidently on recent code.